codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -805,6 +795,35 @@ private InterfaceHttpData findMultipartDisposition() {
         }
     }
 
+    private Attribute getContentDispositionAttribute(String... values) {
+        Attribute attribute;
+        String name = cleanString(values[0]);
+        String value = values[1];
+
+        boolean shouldDecode = false;
+        int last = name.length() - 1;
+        if (name.charAt(last) == '*') {
+            shouldDecode = true;
+            name = name.substring(0, last);
+        }
+
+        // See http://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html
+        if (HttpHeaderValues.FILENAME.contentEquals(name)) {
+            // filename value is quoted string so strip them
+            if (!shouldDecode) {
+                value = value.substring(1, value.length() - 1);
+            } else {
+                String[] split = value.split("''", 2);
```

Comment:
Consider replacing `String#split` with use precompiled `Pattern` constant or a `value.indexOf("''")`.

Reply:
Thank you. I don't have much experience in performant Java string work, so I was following the lead of existing code a [few lines up](https://github.com/dminkovsky/netty/blob/149916d052180dd2b0e0e98598f349454ea200ba/codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java#L707) where you have `value.split("=", 2)`. Does the compiler/runtime optimize this case because it's a unit-length string and therefore effectively a char?

Reply:
Yep. For one-char string a `String.split` in OpenJDK [doesn't use `Pattern`](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/String.java#l2324). But the string `"''"` has two chars.

Reply:
Ah okay thanks. I opted for the precompiled pattern because it seemed simpler.